src/components/ChatApp/ChatApp.react.js

```diff
@@ -11,18 +10,43 @@ import { createMessagePairArray } from '../../utils/formatMessage';
 class ChatApp extends Component {
   static propTypes = {
     actions: PropTypes.object,
+    app: PropTypes.object,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      uuid: '',
+      email: '',
+      userName: '',
+      accessToken: '',
+    };
+  }
+
   componentDidMount() {
+    this.getMessageHistory();
     document.title = 'SUSI.AI Chat - Open Source Artificial Intelligence';
-    // force an update if the URL changes
-    history.listen(() => this.forceUpdate());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { uuid, email, userName, accessToken } = nextProps.app;
```

Comment:
@akshatnitd Sir, I did this to fire this.setState whenever there is a change in app object in the redux store, as the component only re-renders when there is a change in the local state of the component. This would also trigger a rerender for all the children component as you know, and this is what we intend to achieve here as top bar is a child component of the **ChatApp** component. This would also give us a chance to fetch the message store after the login has been completed. I found it to be the most common way of detecting store prop change, so I have used this. Let me know if I missed something.

Reply:
@AakashMallik A component can re-render with prop change also

Reply:
@fragm3 Last time I checked, a component will only re-render if its state changed, or we manually called shouldComponentUbdate().
https://lucybain.com/blog/2017/react-js-when-to-rerender/
If there is some new version update that I am unaware of then do let me know.

Reply:
@fragm3 You were right, the articles got me confused. Yes indeed props will trigger re-render. The purpose of componetWillRecieve Props is to make sure when to trigger the message fetching. So we only need to fetch the messages when the uuid changes ( Login / Logout ). That is the only purpose of componentWillRecieve props here. Thanks for helping me out here.